service/history/workflow/update/registry.go

```diff
@@ -217,11 +217,14 @@ func (r *RegistryImpl) findLocked(ctx context.Context, id string) (*Update, bool
 
 	if info, ok := r.store.GetUpdateInfo(ctx, id); ok {
 		if info.GetCompletedPointer() != nil {
+			outcome, err := r.store.GetUpdateOutcome(ctx, id)
 			// Completed, create the Update object but do not add to registry. this
 			// should not happen often.
-			return newCompleted(id, func(ctx context.Context) (*updatepb.Outcome, error) {
-				return r.store.GetUpdateOutcome(ctx, id)
-			}, withInstrumentation(&r.instrumentation)), true
+			return newCompleted(
+				id,
+				outcomeOrErr{outcome: outcome, err: err},
```

Comment:
nit: This isn't very idiomatic--any way to make this more conventional?

Reply:
Not in Go, no. I had `any` with a comment before and it tripped me up so I made it stronger.